### PR TITLE
Fix sign-compare warnings in tensorflow_common and onnx

### PR DESCRIPTION
**Pull request by panhaiqi**
### Details:
 - *Fix sign-compare warnings in tensorflow_common and onnx*

### Tickets:
 - *97264*

### CODE REVIEW

File: src/frontends/tensorflow_common/src/pass/transpose_sinking.cpp [hunk 1]
```diff
@@ -174,13 +174,13 @@ static void convert_binary_to_default_order(const shared_ptr<Node>& binary,
...
     auto left_rank = get_static_rank(left);
-    if (left_rank < perm_to_def.size() && left.get_partial_shape().is_static()) {
+    if (left_rank < static_cast<int64_t>(perm_to_def.size()) && left.get_partial_shape().is_static()) {
         auto left_shape = left.get_shape();
         left_shape.insert(left_shape.begin(), perm_to_def.size() - left_rank, 1);
 
         auto new_shape = apply_permutation(left_shape, perm_to_def);
         new_node = make_reshape(left, new_shape);
-    } else if (left_rank == perm_to_def.size()) {
+    } else if ((size_t)left_rank == perm_to_def.size()) {
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/pass/transpose_sinking.cpp:183]: please revert this. It was recently fixed already

File: src/frontends/tensorflow_common/src/place.cpp [hunk 2]
```diff
@@ -114,10 +109,9 @@ std::vector<ov::frontend::Place::Ptr> OpPlace::get_consuming_ports() const {
 }
 
 ov::frontend::Place::Ptr OpPlace::get_input_port(const std::string& inputName, int inputPortIndex) const {
-    FRONT_END_GENERAL_CHECK(inputPortIndex >= 0, "inputPortIndex is negative.");
-    size_t input_port_index = static_cast<size_t>(inputPortIndex);
-    FRONT_END_GENERAL_CHECK(input_port_index <= m_input_ports.at(inputName).size(), "inputPortIndex is out of bounds.");
-    return m_input_ports.at(inputName)[input_port_index];
+    FRONT_END_GENERAL_CHECK((size_t)inputPortIndex <= m_input_ports.at(inputName).size(),
+                            "inputPortIndex is out of bounds.");
+    return m_input_ports.at(inputName)[inputPortIndex];
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/place.cpp:114]: please revert this. It was recently fixed already

File: src/frontends/tensorflow_common/src/utils.cpp [hunk 3]
```diff
@@ -241,14 +241,14 @@ ov::OutputVector ov::frontend::tensorflow::translate_convolution_op(const ov::fr
...
+                                                 int min_input_size,
                                                  const std::vector<std::string>& supported_ops) {
     auto op_type = node.get_op_type();
     TENSORFLOW_OP_VALIDATION(node,
                              std::find(supported_ops.begin(), supported_ops.end(), op_type) != supported_ops.end(),
                              op_type + " is not supported for conversion.");
     TENSORFLOW_OP_VALIDATION(node,
-                             node.get_input_size() >= min_input_size,
+                             node.get_input_size() >= (size_t)min_input_size,
                              op_type + " must have at least " + std::to_string(min_input_size) + " inputs.");
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/utils.cpp:252]: please revert this. It was recently fixed already

File: src/frontends/onnx/frontend/src/editor.cpp [hunk 4]
```diff
@@ -175,86 +175,6 @@ void modify_initializer(TensorProto& initializer,
         tensor_type->set_elem_type(initializer.data_type());
     }
 }
-
-bool is_topologically_sorted(const GraphProto& graph) {
```
Comment (mbencer) [inline src/frontends/onnx/frontend/src/editor.cpp:179]: ONNX FE should be topologically sorted according to the standard, but we try to be more user-friendly. Why this change is removed?

**Review by mbencer — CHANGES_REQUESTED** [review]
